code.js

```diff
@@ -1,3 +1,40 @@
+// References:
+// https://www.geeksforgeeks.org/binary-search-in-javascript/ for help
+// https://github.com/COSC3020/binary-search-ZachRenz/blob/main/code.js
+// https://www.youtube.com/watch?v=P3YID7liBug (helped for visualization of algorithm)
+
 function binarySearch(list, element) {
+    let mid;
+    if(list.length == 1){
+        if(list[0] == element){
+            return 0;
+        }
+    }
+    if(list.length % 2 == 0){
```

Comment:
Don't understand this check here. Why does that make a difference?

Reply:
In reality it makes no difference. I started getting in my head about even and odd lists, but the Math.floor() rounds down anyhow so it was unnecessary.